src/client/sandbox/event/message.js

```diff
@@ -210,13 +214,22 @@ export default class MessageSandbox extends SandboxBase {
                 }
             };
 
-            // NOTE: Imitation of a delay for the postMessage method.
-            // We use the same-domain top window
-            // so that the function called by setTimeout is executed after removing the iframe
-            var topSameDomainWindow = getTopSameDomainWindow(this.window);
-            var timeoutId           = nativeMethods.setTimeout.call(topSameDomainWindow, sendFunc, 10);
-
-            this.iframeInternalMsgQueue.push({ timeoutId, sendFunc });
+            if (this.isWindowUnloaded) {
```

Comment:
I would make smthng like 'force' flag for skip calling `_removeInternalMsgFromQueue` method in `sendFunc` and call it if `this.isWindowUnloaded` is true.